nextcord/components.py

```diff
@@ -261,13 +268,274 @@ def to_dict(self) -> SelectMenuPayload:
+        self.max_values: int = data.get("max_values", 1)
+        self.disabled: bool = data.get("disabled", False)
+
+    def to_dict(self) -> MentionableSelectMenuPayload:
+        payload: MentionableSelectMenuPayload = {
+            "type": self.type.value,
+            "custom_id": self.custom_id,
+            "min_values": self.min_values,
+            "max_values": self.max_values,
+            "disabled": self.disabled,
+        }
+
+        if self.placeholder:
+            payload["placeholder"] = self.placeholder
+
+        return payload
+
+
+class ChannelSelectMenu(Component):
+    """Represents a mentionable select menu from the Discord Bot UI Kit.
+
+    A mentionable select menu is functionally the same as a dropdown, however
+    on mobile it renders a bit differently.
+
+    .. note::
+
+        The user constructible and usable type to create a select menu is
+        :class:`nextcord.ui.Select` not this one.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    ----------
+    custom_id: Optional[:class:`str`]
+        The ID of the select menu that gets received during an interaction.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "custom_id",
+        "placeholder",
+        "min_values",
+        "max_values",
+        "disabled",
+        "channel_types",
+    )
+
+    __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
+
+    def __init__(self, data: ChannelSelectMenuPayload):
+        self.type = ComponentType.select
+        self.custom_id: str = data["custom_id"]
+        self.placeholder: Optional[str] = data.get("placeholder")
+        self.min_values: int = data.get("min_values", 1)
+        self.max_values: int = data.get("max_values", 1)
+        self.disabled: bool = data.get("disabled", False)
+        self.channel_types: List[ChannelType] = [
+            ChannelType(t) for t in data.get("channel_types", [])
+        ]
+
+    def to_dict(self) -> ChannelSelectMenuPayload:
+        payload: ChannelSelectMenuPayload = {
+            "type": self.type.value,
+            "custom_id": self.custom_id,
+            "min_values": self.min_values,
+            "max_values": self.max_values,
+            "disabled": self.disabled,
+        }
+
+        if self.placeholder:
+            payload["placeholder"] = self.placeholder
+
+        if self.channel_types:
+            payload["channel_types"] = [t.value for t in self.channel_types]
+
         return payload
```

Comment:
Is there a reason all these classes reimplement the same thing vs using inheritance from a common base?

Reply:
Kinda to make it also possible to pass some integer